geode-core/src/main/java/org/apache/geode/pdx/internal/PeerTypeRegistration.java

```diff
@@ -350,23 +350,25 @@ private void releaseUDPMessaging(boolean release) {
   public int defineType(PdxType newType) {
     verifyConfiguration();
     Integer existingId = typeToId.get(newType);
+
     if (existingId != null) {
       return existingId;
     }
     lock();
     try {
-      int id = getExistingIdForType(newType);
-      if (id != -1) {
-        return id;
+      if (typeToId.isEmpty()) {
```

Comment:
Add a unit test for PeerTypeRegistration and tests in it for the methods you are changing.

Reply:
The while class has no junit tests. We always use dunit/integration test to cover. And this fix is also covered by dunit test. We will add one more integration test. 

According to Darrel's suggestion, I filed a new ticket GEODE-7034 to refactor the class to be junit testable.